File: src/types/prerequisites/single/AnimistPowerPrerequisite.ts
import * as DB from "tsondb/schema/dsl"
import { AnimistPowerIdentifier } from "../../_Identifier.js"
import { DisplayOption } from "../DisplayOption.js"

export const AnimistPowerPrerequisite = DB.TypeAlias(import.meta.url, {
  name: "AnimistPowerPrerequisite",
  comment: "Requires a specific animist power to be on a minimum value.",
  type: () =>
    DB.Object({
      id: DB.Required({
        comment: "The animist power’s identifier.",
        type: AnimistPowerIdentifier(),
      }),
      level: DB.Optional({
        comment: "The level to which the minimum value applies.",
        type: DB.Integer({ minimum: 2 }),
      }),
      value: DB.Required({
        comment: "The required minimum value.",
        type: DB.Integer({ minimum: 0 }),
      }),
      display_option: DB.Optional({
        type: DB.IncludeIdentifier(DisplayOption),
      }),
    }),
})

File: Sources/OptolithDatabaseSchema/Types/prerequisites/single/AnimistPowerPrerequisite.swift
import FileDB

/// Requires a specific animist power to be on a minimum value.
@Embedded
public struct AnimistPowerPrerequisite {

  /// The animist power’s identifier.
  @Relationship(AnimistPower.self)
  let id: AnimistPower.ID

  /// The level to which the minimum value applies.
  @Minimum(2)
  let level: Int?

  /// The required minimum value.
  @Minimum(0)
  let value: Int

      let display_option: DisplayOption?
  }
